SLING-8854 In-file and In-memory queue-providers for Forward …

actinium15 opened the pull request:
…Distribution report incorrect value for processing "Attempts"

* added a 'default' method to DistributionQueue SPI for
  DistributionQueueProcessor (or rather, any caller of
  DistributionQueue.getHead) to express the intent behind the call.

actinium15 committed:
SLING-8854 In-file and In-memory queue-providers for Forward Distribution report incorrect value for processing "Attempts"
* added a 'default' method to DistributionQueue SPI for
  DistributionQueueProcessor (or rather, any caller of
  DistributionQueue.getHead) to express the intent behind the call.

tmaret reviewed (changes requested):
See https://issues.apache.org/jira/browse/SLING-8854?focusedCommentId=16981515&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16981515

actinium15 committed:
SLING-8854 Keeping SonarQube happy (take #2)
* the deleted annotation _seems to be_ the only line marked 'duplicate'
  in the new code. This is definitely not the right thing to do, but
  adding this commit just to see if the real problem lies somewhere else

actinium15 committed:
SLING-8854 Keeping SonarQube happy

tmaret commented:
Closing as #26 has been merged. This approach would have extended the API. The API extension, `recordProcessingAttempt` signature, is not generic enough on the `DistributionQueue` API, the attempt count is exposed via the `DistributionQueueItemStatus` API.
